lib/include/jxl/decode.h

```diff
@@ -238,6 +243,39 @@ typedef enum {
    * image and always before JXL_DEC_FULL_IMAGE.
    */
   JXL_DEC_JPEG_RECONSTRUCTION = 0x2000,
+
+  /** Informative event by JxlDecoderProcessInput: The header of a box of the
+   * BMFF container format is decoded. The following API functions related to
+   * boxes can be used after this event:
+   * @see JxlDecoderSetBoxBuffer and JxlDecoderReleaseBoxBuffer: set and release
+   * a buffer to get the uncompressed box data.
+   * @see JxlDecoderGetBoxType get the 4-character uncompressed box typename.
+   * @see JxlDecoderBoxTypeEquals to compare with the 4-character uncompressed
+   * box typename.
+   * @see JxlDecoderIsBoxCompressed to query whether the box was stored as a
+   * compressed box of type "brob", or directly uncompressed.
+   * @see JxlDecoderGetBoxTypeRaw get the 4-character raw, "brob" if compressed,
+   * box typename.
+   * @see JxlDecoderBoxTypeRawEquals to compare with the raw box typename.
+   * @see JxlDecoderGetBoxSizeRaw to get the raw compressed box size.
+   * @see JxlDecoderSetBoxModeRaw to configure whether to get the box
+   * data uncompressed, or possibly compressed. Getting the raw box contents
+   * returns the same data and type unless JxlDecoderIsBoxCompressed is true. In
+   * the compressed box case, the raw typename is "brob" but the underlying
+   * typename can still be requested, and the data is brotli compressed. To get
+   * metadata, use the non-raw box mode. To perform more primitive operations
+   * with boxes, use the raw box mode and the non-raw type if needed. The buffer
+   * set with JxlDecoderSetBoxBuffer must be set again for each next box, or can
+   * be left unset to skip outputting this box. The output buffer contains the
+   * full box data when the next JXL_DEC_BOX event occurs or when
```

Comment:
What about unrelated events, say JXL_DEC_FRAME?

Reply:
Those work independently, and it's not specified if you get box data before, after, or during the decoding of pixels. You could get a box in the middle of a frame (because a frame could be split over multiple partial boxes), or you could get a frame in the middle of a box (if it contains multiple), but I don't think the other events need to be mentioned, specifying this happens at the next box or the end should be complete.